shortfin/python/shortfin_apps/sd/components/generate.py

```diff
@@ -95,12 +96,8 @@ async def run(self):
 
             # TODO: stream image outputs
             logging.debug("Responding to one shot batch")
-            out = io.BytesIO()
-            result_images = [p.result_image for p in gen_processes]
-            for idx, result_image in enumerate(result_images):
-                out.write(result_image)
-                # TODO: save or return images
-                logging.debug("Wrote images as bytes to response.")
-            self.responder.send_response(out.getvalue())
+            response_data = {"images": [p.result_image for p in gen_processes]}
```

Comment:
A couple of notes to self here: a) Need to get tracing tags up to this level so we can see specific overheads, b) need to add a general purpose IO task queue so we can easily offload stuff like this to a side-thread